common/flatpak-dir.c

```diff
@@ -8553,6 +8566,39 @@ flatpak_dir_prune (FlatpakDir   *self,
 
 }
 
+gboolean
+flatpak_dir_update_summary (FlatpakDir   *self,
+                            GCancellable *cancellable,
+                            GError      **error)
+{
+  g_autoptr(GError) local_error = NULL;
+  g_auto(GLnxLockFile) lock = { 0, };
+
+  if (error == NULL)
+    error = &local_error;
+
+  if (flatpak_dir_use_system_helper (self, NULL))
+    {
+      const char *installation = flatpak_dir_get_id (self);
+
+      return flatpak_dir_system_helper_call_update_summary (self,
+                                                            installation ? installation : "",
+                                                            cancellable,
+                                                            error);
+    }
+
+  if (!flatpak_dir_ensure_repo (self, cancellable, error))
+    return FALSE;
+
+  /* Keep a shared repo lock to avoid prunes removing objects we're relying on
+   * while generating the summary. */
+  if (!flatpak_dir_repo_lock (self, &lock, LOCK_SH, cancellable, error))
+    return FALSE;
+
+  g_debug ("Updating summary");
+  return ostree_repo_regenerate_summary (self->repo, NULL, cancellable, error);
```

Comment:
Passing `NULL` to `additional_metadata` here will wipe out any additional metadata which was stored in the `summary` when it was last generated. Is that OK?

Reply:
I don’t think we want the caller to be able to provide additional metadata (certainly not if the caller is unprivileged), but perhaps we’d want to preserve additional metadata from one version of the `summary` to the next? I’m not sure.

Reply:
Yeah I don't see why that would be a problem. `ostree_repo_pull_with_options` (which is used by `create-usb`) uses the summary as a list of refs. Flatpak uses the additional metadata when it's a remote summary whereas this is the local summary. We're also already doing the same thing (leaving out additional metadata) for the child repo during flatpak pulls, and for the repo on destination USB drives.

Reply:
:+1: